src/pages/Swap/Swap.tsx

```diff
@@ -1,105 +1,107 @@
+import { FormInstance } from 'antd';
+import { useForm } from 'antd/es/form/Form';
+import React, { useEffect } from 'react';
 
-import { SubmitButton } from '../../components/SubmitButton/SubmitButton';
-import { TokenInput } from '../../components/TokenInput/TokenInput';
-import { TokenListModal } from '../../components/TokenListModal/TokenListModal';
+import {
+  ActionForm,
+  ActionFormStrategy,
+} from '../../components/common/ActionForm/ActionForm';
+import {
+  TokenControlFormItem,
+  TokenControlValue,
+} from '../../components/common/TokenControl/TokenControl';
+import { FormPageWrapper } from '../../components/FormPageWrapper/FormPageWrapper';
 import {
   Box,
   Button,
-  Col,
+  Flex,
   HistoryOutlined,
-  Modal,
-  Row,
   SettingOutlined,
   SwapOutlined,
   Typography,
 } from '../../ergodex-cdk';
 
-export const Swap: React.FC = () => {
-  const [fromValue, setFromValue] = useState('');
-  const [fromTokenName, setFromTokenName] = useState('ERG');
-  const [fromTokenBalance] = useState(0);
-  const [fromTokenPrice] = useState(335);
-  const [toValue, setToValue] = useState('');
-  const [toTokenName, setToTokenName] = useState('');
-  const [toTokenBalance] = useState(0);
-  const [toTokenPrice] = useState(335);
+interface SwapFormModel {
+  readonly from: TokenControlValue;
+  readonly to: TokenControlValue;
+}
+
+class SwapStrategy implements ActionFormStrategy {
+  actionButtonCaption(): React.ReactNode {
+    return 'Swap';
+  }
+
+  getInsufficientTokenForFee(form: FormInstance): string | undefined {
+    return undefined;
+  }
+
+  getInsufficientTokenForTx(form: FormInstance): string | undefined {
+    return 'ERG';
+  }
+
+  isAmountNotEntered(form: FormInstance<SwapFormModel>): boolean {
+    const value = form.getFieldsValue();
+
+    return !value.from?.amount?.value || !value.to?.amount?.value;
+  }
+
+  isTokensNotSelected(form: FormInstance<SwapFormModel>): boolean {
+    const value = form.getFieldsValue();
+
+    return !value.to?.token || !value.from?.token;
+  }
+
+  request(form: FormInstance): Promise<any> {
+    return Promise.resolve(undefined);
+  }
+
+  isLiquidityInsufficient(form: FormInstance<any>): boolean {
+    return false;
+  }
+}
+
+export const Swap = () => {
+  const [form] = useForm<SwapFormModel>();
+  const swapStrategy = new SwapStrategy();
 
-  const handleSelectFromToken = (onSelectChanged: (name: string) => void) => {
-    Modal.open(
-      ({ close }) => (
-        <TokenListModal close={close} onSelectChanged={onSelectChanged} />
-      ),
-      { title: 'Select a token' },
-    );
+  const swapTokens = () => {
+    form.setFieldsValue({
+      from: form.getFieldsValue().to,
+      to: form.getFieldsValue().from,
+    });
   };
 
   return (
-    <Row align="middle" justify="center">
-      <Col span={7}>
-        <Box className="swap" formWrapper borderRadius="l" padding={6}>
-          <Row bottomGutter={6}>
-            <Col span={18}>
+    <FormPageWrapper width={480}>
```

Comment:
Is it reasonable to rename this component to DexFormPageWrapper?
Coz we will have default forms with default validations too.

Reply:
I think, we have not only form pages in this style. So, can we call component Page/CardPage ?